### DISCUSSION TITLE
Obsidian web has stopped working on Mac

### DISCUSSION THREAD

**Discussion by huangjingyun8** [coddingtonbear/obsidian-web#201] [Bug Report]
### Describe the bug

Since the update of the Obsidian Local REST API two days ago, Obsidian web has stopped working on Mac. However, it works fine on Windows, and I don't understand why.

### To Reproduce

2. Click on '...'
3. Scroll down to '...'

### Expected Behavior

<img width="910" alt="image" src="https://github.com/coddingtonbear/obsidian-web/assets/56347021/b1af605d-c2ea-4bea-8ab1-76555a649e66">

### Operating System

mac

### Browser

chrome

### Browser Version

126.0.6478.62

**Comment by coddingtonbear** [comment]
The first thing that comes to mind is maybe Obsidian Web doesn't have access to that host.  Could you follow the following steps and tell me what you see?

1. Right-click on the extension icon.
2. Select 'Manage Extension'.
3. From the page that is opened, look for a section labeled "Site Access" and send me a screenshot of it.

What it looks like for _me_ is:

![image](https://github.com/coddingtonbear/obsidian-web/assets/527661/1f8ad091-bff9-4d7e-b816-387970686528)

but I suspect it's different in your case?

**Reply by huangjingyun8** [reply]
<img width="776" alt="微信图片_20240620094214" src="https://github.com/coddingtonbear/obsidian-web/assets/56347021/c8fc1207-4664-4bbe-b304-c407ab62346c">

**Reply by coddingtonbear** [reply]
I'm afraid I don't speak mandarin; so I have no idea how to interepret that.  Can you tell me what this part means, when translated to English?

![image](https://github.com/coddingtonbear/obsidian-web/assets/527661/f588e453-05df-4843-b029-ca2aafce2b9a)

**Reply by huangjingyun8** [reply]
on all sites

**Reply by coddingtonbear** [reply]
that looks correct; could you attach a “bug report export” using al the red bug-shaped button in the upper-right corner of the extension options page?

**Reply by huangjingyun8** [reply]
[
[obsidian-local-rest-api.bug_report.json](https://github.com/user-attachments/files/15908049/obsidian-local-rest-api.bug_report.json)
](url)

**Reply by coddingtonbear** [reply]
The export shows that you’re connecting to `https://127.0.0.1:27124/`, which means you need to have the certificate installed for things to work.  

Either:

1. Install the certificate (see Obsidian Local REST API’s settings) or
2. Make sure you have turned on the “non-encrypted” service in Obsidian Local REST API, and change the API URL in Obsidian Web to match the Non-encrypted API URL shown in Obsidian Local REST API’s settings.

& In general, it’s a good idea to make sure Obsidian Local REST API is up-to-date in Obsidian.

**Reply by huangjingyun8** [reply]
![image](https://github.com/coddingtonbear/obsidian-web/assets/56347021/56daf030-3bf0-4ae8-a9d9-289a23ca4f2e)

I'm not sure if there are additional settings needed, as I've never touched anything before. Do I need to download a separate file to install a certificate on a Mac? I'm a beginner.

**Reply by huangjingyun8** [reply]
Unable to connect to Obsidian: TypeError: Failed to fetch. This error occurs when Obsidian Web does not have permission to access this hostname, or when Obsidian Local REST API is running in secure-only mode and your browser does not trust its certificate. Make sure you have granted permission for this host, and either enable insecure mode from Obsidian Local REST API's settings panel, or see the settings panel for instructions regarding where to acquire the certificate you need to configure your browser to trust.

**Reply by coddingtonbear** [reply]
Yes, to resolve that, see my message here: https://github.com/coddingtonbear/obsidian-web/discussions/201#discussioncomment-9827834

**Reply by huangjingyun8** [reply]
I have chosen option 2, which is non-encrypted, and the issue has been temporarily resolved. Thank you very much."

**Reply by coddingtonbear** [reply]
No problem!  To use the encrypted version, you'll have to figure out how to install the certificate & configure your operating system to trust it.  I'm not sure what the process for trusting a certificate would be on a mac, but if you do figure it out a way of doing that, other folks who ware also on a mac might find it helpful if you were to add some instructions to the wiki here: https://github.com/coddingtonbear/obsidian-web/wiki/How-do-I-get-my-browser-trust-my-Obsidian-Local-REST-API-certificate%3F -- I personally only have direct experience with this sort of thing on Linux.

**Reply by OLH21** [reply]
Thanks for the explanation that solve my problem here too, though it used to work, does an update broke something ?

**Comment by JessicaSprague** [comment]
I was receiving this exact error today, I'm using Chrome on Mac Sonoma and it took awhile (couple hours tbh) to figure out how to approximate the Windows instructions, which get pretty different from Mac. I don't know how to add a page to the Wiki for other Mac users, so I thought I would just sum up here and yall can either tell me how to add them, or someone else can?

I noted that Mac OS aggressively dissuades from adding certificates manually, so it probably seems like you're doing something terrible and world-ending, but then it's like oh, ok that's fine. 

1. Download the certificate, and navigate TO the certificate in Finder (don't try to open it w Chrome via the download history) 
2. Double-click the certificate file, and add the certificate to the System Keychain (you'll need to log in with your Mac password)
3. Under Trust, set to Always Trust. 

<img width="1018" alt="ObsidianScreenshot" src="https://github.com/user-attachments/assets/5f8b8ddf-bfcb-4b02-95e7-66edc9a848e8">

Now check the browser extension by right-clicking it in Chrome, and the little red shield of NO! next to your API key should have turned into a friendly green checkmark of YES! and that particular error should be gone.
